src/index.ts

```diff
@@ -218,50 +193,37 @@ export class IDX {
         const [key, ref] = list[cursor++]
         const doc = await this._loadDocument(ref)
         return { done: false, value: { key, ref, content: doc.content } }
-      }
+      },
     }
   }
 
-  async _getIDXDoc(did: string): Promise<Doctype | null> {
-    let rootId
-    rootId = this._didCache[did]
-    if (rootId === null) {
-      // If explicitly `null` the DID has already been resolved and does not support IDX
-      return null
-    }
-
-    if (rootId == null) {
-      // If undefined try to resolve the DID and check support for IDX
-      rootId = await this.getIDXDocID(did)
-      this._didCache[did] = rootId
-      if (rootId == null) {
-        return null
-      }
-    }
-
-    const doc = await this._loadDocument(rootId)
+  async _getIDXDoc(did: string): Promise<Doctype> {
+    const doc = await this._ceramic.createDocument(
+      'tile',
+      {
+        deterministic: true,
+        metadata: { controllers: [did], family: 'IDX', schema: schemas.IdentityIndex },
```

Comment:
We should **not** set the schema here. The schema should be set the first time we update the document. The reason for this is that if we update the schema we would have to be aware of which version of the schema to use here, which is not good.

Reply:
Good point, thanks for spotting this!